config/config.go

```diff
@@ -0,0 +1,145 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package config provides types for specifying the expected configuration of a
+// Conduit plugin (connector or processor). It also provides utilities to
+// validate the configuration based on the specifications.
+package config
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Config is a map of configuration values. The keys are the configuration
+// parameter names and the values are the configuration parameter values.
+type Config map[string]string
+
+// Sanitize removes leading and trailing spaces from all keys and values in the
+// configuration.
+func (c Config) Sanitize() Config {
+	for key, val := range c {
+		key = strings.TrimSpace(key)
+		val = strings.TrimSpace(val)
+		c[key] = val
+	}
+	return c
+}
+
+// ApplyDefaults applies the default values defined in the parameter
+// specifications to the configuration. If a parameter is not present in the
+// configuration, the default value is applied.
+func (c Config) ApplyDefaults(params Parameters) Config {
+	for key, param := range params {
+		if strings.TrimSpace(c[key]) == "" {
```

Comment:
What if a value has intentionally been set to an empty string? Maybe it's safer to check for key existence? `

Reply:
Some time at the start of Conduit we decided not to make a distinction between missing and empty configuration values, it simplifies these kinds of checks. Connector configurations would get more complex if we made the distinction, we would need checks to distinguish empty and missing configuration fields. So it was a conscious decision to make it easier on us. Also, given that we had this behavior in connectors thus far and we didn't see a need for this, I wouldn't want to change it without a good reason.

Reply:
With that decision, that sounds good to me! It would be good if we had it somewhere documented, I think it's an important one.

Reply:
https://github.com/ConduitIO/conduit-connector-sdk/pull/110